scaleway/resource_compute_instance_server.go

```diff
@@ -0,0 +1,360 @@
+package scaleway
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/utils"
+)
+
+func resourceScalewayComputeInstanceServer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalewayComputeInstanceServerCreate,
+		Read:   resourceScalewayComputeInstanceServerRead,
+		Update: resourceScalewayComputeInstanceServerUpdate,
+		Delete: resourceScalewayComputeInstanceServerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The name of the server",
+			},
+			"image": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "The base image of the server", // TODO: add in doc example with UUID
+				ValidateFunc: validationUUID(),
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The instance type of the server", // TODO: link to scaleway pricing in the doc
+			},
+			"tags": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The tags associated with the server",
+			},
+			"security_group_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The security group the server is attached to",
+			},
+			"root_volume": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"size_in_gb": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+							ForceNew: true, // TODO: don't force new but stop server and create new volume instead
+						},
+						"delete_on_termination": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  true,
```

Comment:
Do you really want this to default to true? It's not the behaviour of the console. And it can make people lose data.

Reply:
It is the behavior of the console.